src/Common/src/CoreLib/System/SpanHelpers.Byte.cs

```diff
@@ -494,15 +494,15 @@ public static unsafe int IndexOfAny(ref byte searchSpace, byte value0, byte valu
             return (int)(byte*)(index + 7);
         }
 
-        public static unsafe int IndexOfAny(ref byte searchSpace, byte value0, byte value1, byte value2, int length)
+        public static unsafe int IndexOfAny(ref byte searchSpace, byte value0, byte value1, byte value2, nuint length)
         {
             Debug.Assert(length >= 0);
 
             uint uValue0 = value0; // Use uint for comparisons to avoid unnecessary 8->32 extensions
             uint uValue1 = value1; // Use uint for comparisons to avoid unnecessary 8->32 extensions
             uint uValue2 = value2; // Use uint for comparisons to avoid unnecessary 8->32 extensions
             IntPtr index = (IntPtr)0; // Use UIntPtr for arithmetic to avoid unnecessary 64->32->64 truncations
-            IntPtr nLength = (IntPtr)(uint)length;
+            IntPtr nLength = (IntPtr)length;
```

Comment:
We should use nuint instead of IntPtr in the implementation instead of the hacky combination of IntPtrs and pointers.